### PR TITLE
fates hydro refactors

### PR DESCRIPTION
**Pull request by rgknox**
### Description of changes

This is a synced PR with NGEET/fates [611](https://github.com/NGEET/fates/pull/611).  The couplet is designed to improve stability of FATES hydro and get tests working.

### Specific notes

Contributors other than yourself, if any:
@bchristo @xuchongang @pnlfang @ckoven @JunyanDing and others

CTSM Issues Fixed (include github issue #):

Are answers expected to change (and if so in what way)?

FATES hydro answers should change.  Non hydro should not change.

Any User Interface Changes (namelist or namelist defaults changes)?

Two deprecated variables were removed from the FatesHydro test environment.

Testing performed, if any:
(List what testing you did to show your changes worked as expected)
(This can be manual testing or running of the different test suites)
(Documentation on system testing is here: https://github.com/ESCOMP/ctsm/wiki/System-Testing-Guide)
(aux_clm on cheyenne for intel/gnu and izumi for intel/gnu/nag/pgi is the standard for tags on master)

TBD

**NOTE: Be sure to check your Coding style against the standard:**
https://github.com/ESCOMP/ctsm/wiki/CTSM-coding-guidelines

### CODE REVIEW

File: src/utils/clmfates_interfaceMod.F90 [hunk 1]
```diff
@@ -2215,7 +2231,16 @@ subroutine ComputeRootSoilFlux(this, bounds_clump, num_filterc, filterc, &
...
        nlevsoil = this%fates(nc)%bc_in(s)%nlevsoil
+
+       ! This is the water removed from the soil layers by roots (or added)
        waterflux_inst%qflx_rootsoi_col(c,1:nlevsoil) = this%fates(nc)%bc_out(s)%qflx_soil2root_sisl(1:nlevsoil)
+
+       ! This is the total amount of water transferred to surface runoff
+       ! (this is generated potentially from supersaturating soils
+       ! (currently this is unnecessary)
+       ! waterflux_inst%qflx_drain_vr_col(c,1:nlevsoil) = this%fates(nc)%bc_out(s)%qflx_ro_sisl(1:nlevsoil)
+
```
Comment (glemieux) [inline src/utils/clmfates_interfaceMod.F90:2242]: Is this commented out for future potential fix or addition?

File: src/utils/clmfates_interfaceMod.F90 [hunk 2]
```diff
@@ -481,7 +482,7 @@ subroutine init(this, bounds_proc )
          call this%init_soil_depths(nc)
          
          if (use_fates_planthydro) then
-            call InitHydrSites(this%fates(nc)%sites,this%fates(nc)%bc_in,numpft_fates)
```
Comment (glemieux) [inline src/utils/clmfates_interfaceMod.F90:484]: Looking at the change in the fates PR it looks like `numpft` might still be needed for `InitHydrSites`?

File: cime_config/testdefs/testmods_dirs/clm/FatesHydro/user_nl_clm [hunk 3]
```diff
@@ -1,5 +1,9 @@
 hist_empty_htapes = .true.
 use_fates_planthydro= .true.
-hist_fincl1       = 'FATES_ERRH2O_SCPF','FATES_TRAN_SCPF','FATES_ROOTUPTAKE_SCPF','FATES_ROOTUPTAKE01_SCPF',
-'FATES_SAPFLOW_COL_SCPF','FATES_ITERH1_COL_SCPF','FATES_ATH_COL_SCPF','FATES_TTH_COL_SCPF',
-'FATES_STH_COL_SCPF','FATES_LTH_COL_SCPF','FATES_AWP_COL_SCPF','FATES_BTRAN_COL_SCPF'
+hist_fincl1       = 'FATES_ERRH2O_SCPF','FATES_TRAN_SCPF',
```
Comment (glemieux) [inline cime_config/testdefs/testmods_dirs/clm/FatesHydro/user_nl_clm:3]: Looks like this is just a subset of the total history variables added in the fates PR yes?  Just want to confirm.

File: src/utils/clmfates_interfaceMod.F90 (hunk 1 above)
Comment (rgknox) [reply src/utils/clmfates_interfaceMod.F90:2242]: yes, exactly

File: src/utils/clmfates_interfaceMod.F90 (hunk 2 above)
Comment (rgknox) [reply src/utils/clmfates_interfaceMod.F90:484]: looks consistent to me: https://github.com/NGEET/fates/blob/320a8fc0875e228df4133544f122c83844d8720e/biogeophys/FatesPlantHydraulicsMod.F90#L1242

File: cime_config/testdefs/testmods_dirs/clm/FatesHydro/user_nl_clm (hunk 3 above)
Comment (rgknox) [reply cime_config/testdefs/testmods_dirs/clm/FatesHydro/user_nl_clm:3]: yeah, this test focuses just on the hydraulics diagnostic variables.  All of the new ones are in here.

**Comment by glemieux** [comment]
Looks good

**Pull request merged**